table/block_based/block_based_table_reader_sync_and_async.h

```diff
@@ -44,17 +44,16 @@ DEFINE_SYNC_AND_ASYNC(void, BlockBasedTable::RetrieveMultipleBlocks)
     size_t idx_in_batch = 0;
     for (auto mget_iter = batch->begin(); mget_iter != batch->end();
          ++mget_iter, ++idx_in_batch) {
-      BlockCacheLookupContext lookup_data_block_context(
```

Comment:
RetrieveMultipleBlocks is not intended to lookup in the block cache. The cache has already been checked when this function is called. As I suggest with my XXX comment, this code appears to be checking the block cache an extra time unnecessarily. I didn't want to fix that right now, at risk of changing too much in this one PR. Nevertheless, the code was previously collecting tracing info in the lookup_context and always throwing it away. At least we can instead use lookup_context=nullptr to avoid collecting the tracing info. This is one small step in the right direction.